lib/rihanna/job_dispatcher.ex

```diff
@@ -74,28 +74,26 @@ defmodule Rihanna.JobDispatcher do
 
   def handle_info({:DOWN, ref, :process, _pid, reason}, state = %{pg: pg, working: working}) do
     {job, working} = Map.pop(working, ref)
+    job_failure(job, reason, pg)
+    {:noreply, Map.put(state, :working, working)}
+  end
 
+  defp job_failure(%{id: id, term: {job_module, args}}, reason, pg) do
     # NOTE: Do we need to demonitor here?
+    job_module.after_error(reason, args)
```

Comment:
I am surprised all the tests pass here. What if `job_module` does not implement `after_error`?